src/Symfony/Component/Yaml/Escaper.php

```diff
@@ -86,4 +86,30 @@ public static function escapeWithSingleQuotes($value)
     {
         return sprintf("'%s'", str_replace('\'', '\'\'', $value));
     }
+
+    /**
+     * Determines if a PHP value contains any single characters that would cause
+     * the value to require single quoting in YAML.
+     *
+     * @param  string $value A PHP value
+     * @return bool          True if the value would require single quotes.
+     */
+    private static function containsCharRequiresSingleQuoting($value)
+    {
+        return preg_match('/[ \s \' " \: \{ \} \[ \] , & \* \# \?] | \A[ \- ? | < > = ! % @ ` ]/x', $value);
+    }
+
+    /**
+     * Determines if a PHP value is entirely composed of a value that would
+     * require single quoting in YAML.
+     *
+     * @param  string $value A PHP value
+     * @return bool          True if the value would require single quotes.
+     */
+    private static function isValueRequiresSingleQuoting($value)
```

Comment:
Actually, I'm wondering if we really need those 2 private methods. What about inlining them with the right comments instead? Especially as they can be called quite a lot.